File: Algorithm/leetcode_lcp67.go
/**
 * File              : leetcode_lcp67.go
 * Author            : Andy
 * Date              : 2022.10.12
 * Last Modified Date: 2022.10.12
 * Last Modified By  : Andy
 */
package expandBinaryTree

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func expandBinaryTree(root *TreeNode) *TreeNode {
	if root == nil {
		return nil
	}

	var dfs func(node *TreeNode) *TreeNode

	dfs = func(node *TreeNode) *TreeNode {
		if node == nil {
			return nil
		}
		if node.Left != nil {
			// 注意这里后面是结构体，要用{}
			// 并且是用指针
			Left := &TreeNode{Val: -1, Left: node.Left}
			node.Left = Left
			dfs(node.Left.Left)
		}

		if node.Right != nil {
			Right := &TreeNode{Val: -1, Right: node.Right}
			node.Right = Right
			dfs(node.Right.Right)
		}
		return node
	}
	return dfs(root)
}

File: Algorithm/leetcode_lcp67.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# File              : leetcode_lcp67.py
# Author            : Andy
# Date              : 2022.10.12
# Last Modified Date: 2022.10.12
# Last Modified By  : Andy


class Solution:

    def expandBinaryTree(self, root):
        if not root:
            return

        def dfs(node):
            if not node:
                return

            if node.left:
                node.left = TreeNode(-1, left=node.left)
                dfs(node.left.left)

            if node.right:
                node.right = TreeNode(-1, right=node.right)
                dfs(node.right.right)

            return node

        return dfs(root)
